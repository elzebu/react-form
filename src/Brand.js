import React from 'react';

const Brand = (props) => (
    <div className="brand row">
        <div className="col">
            <h3><img src={props.data.src} alt="" height="50" /> {props.data.name}</h3>
        </div>
        <div className="col text-right">
            {props.click ?
                <button onClick={props.click}>supprimer</button>
                : null
            }
        </div>
    </div>
)

export default Brand;