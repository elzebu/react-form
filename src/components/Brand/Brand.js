import React from 'react'

import PropTypes from 'prop-types'

class Brand extends React.Component {
    constructor(props) {
        super(props)
        console.log(`[Brand] ${props.data.name} Constructor`)
        this.state = {}
    }

    componentDidMount() {
        console.log(`[Brand] ${this.props.data.name} Component Did Mount`)
    }

    static getDerivedStateFromProps(props, state) {
        console.log(`[Brand] ${props.data.name} getDerivedStateFromProps`, props, state)
        return state;
    }

    shouldComponentUpdate() {
        console.log(`[Brand] ${this.props.data.name} shouldComponentUpdate`)
        return true;
    }

    componentWillUnmount() {
        console.log(`[Brand] ${this.props.data.name} component will unmount`)

    }

    render() {
        return (
            <div className="brand row">
                <div className="col">
                    <h3><img src={this.props.data.src} alt="" height="50" /> {this.props.data.name}</h3>
                </div>
                <div className="col text-right">
                    {this.props.click ?
                        <button onClick={this.props.click}>supprimer</button>
                        : null
                    }
                </div>
            </div>
        )
    }
}

Brand.propTypes = {
    data: PropTypes.shape({
        src: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
    })
}

export default Brand;