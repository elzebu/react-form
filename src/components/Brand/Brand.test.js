import React from 'react'
import { create } from 'react-test-renderer'

import Brand from './Brand'

it('match previous snapshot', () => {
    const component = create(
        <Brand data={{src:'toto', 'name': 'test'}} />,
    )
    let brand = component.toJSON()
    expect(brand).toMatchSnapshot()
})
