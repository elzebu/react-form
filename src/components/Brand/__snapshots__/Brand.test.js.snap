// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match previous snapshot 1`] = `
<div
  className="brand row"
>
  <div
    className="col"
  >
    <h3>
      <img
        alt=""
        height="50"
        src="toto"
      />
       
      test
    </h3>
  </div>
  <div
    className="col text-right"
  />
</div>
`;
